vim-patch:8.1.{2235,2236}

janlazo opened the pull request.

janlazo committed:
vim-patch:8.1.2235: "C" with 'virtualedit' set does not include multi-byte char

Problem:    "C" with 'virtualedit' set does not include multi-byte char.
Solution:   Include the whole multi-byte char. (Nobuhiro Takasaki,
            closes vim/vim#5152)
https://github.com/vim/vim/commit/77ccc00340ed2598f7aa09716217e878665964fa

mhinz commented:
Just one slightly related note: I think it would be better to manually remove the `closes vim/vim#...` lines from the commit, because otherwise we spam the Vim project with references from our issues.

janlazo commented:
@mhinz 

```diff
diff --git a/scripts/vim-patch.sh b/scripts/vim-patch.sh
index 2a04805..b0b0161 100755
--- a/scripts/vim-patch.sh
+++ b/scripts/vim-patch.sh
@@ -160,7 +160,7 @@ assign_commit_details() {
 
   vim_commit_url="https://github.com/vim/vim/commit/${vim_commit}"
   vim_message="$(git -C "${VIM_SOURCE_DIR}" log -1 --pretty='format:%B' "${vim_commit}" \
-      | sed -e 's/\(#[0-9]\{1,\}\)/vim\/vim\1/g')"
+      | sed -e 's/\(#[0-9]\{1,\}\)//g')"
   if [[ ${munge_commit_line} == "true" ]]; then
     # Remove first line of commit message.
     vim_message="$(echo "${vim_message}" | sed -e '1s/^patch /vim-patch:/')"
```

janlazo committed:
vim-patch:8.1.2236: ml_get error if pattern matches beyond last line

Problem:    Ml_get error if pattern matches beyond last line.
Solution:   Adjust position if needed. (Christian Brabandt, closes )
https://github.com/vim/vim/commit/bb26596242fa7db477e2cd706dd99f9a426b5f71

janlazo committed:
quickfix: fix pvs/v547 error

janlazo committed:
syntax: zero-init local structs

Fix https://neovim.io/doc/reports/clang/report-ee5dbd.html#EndPath
